### DISCUSSION TITLE
Resetting stepper motor to initial position

### DISCUSSION THREAD

**Discussion by PiezaKake** [bonsai-rx/bonsai#1316] [Q&A] [answered]
Hi,

I am new to Bonsai, and I started using it for a gap-crossing apparatus for mice. The gap between two platforms is controlled by a stepper motor connected to an Arduino board that moves one of the platforms. I start with a gap of 3 cm and incrementally increase it by the same amount every 5 min. I have set up a counter that counts how many times the gap has increased. Once the gap has become to large for the mouse to cross within 5 min, I want the stepper motor to take the amount of steps indicated by the counter to the opposite direction, so that the gap is reset to 3 cm again. Suggestions?

**Comment by bruno-f-cruz** [comment]
Can you share your current workflow? 
Thanks.

**Reply by PiezaKake** [reply]
> Can you share your current workflow? Thanks.

Thanks for the prompt reply. I am expanding on the work of a colleague, but I am still trying to find my way around the workflow. Item2 is the number of times the mouse has crossed the gap. If it is not equal to 0, the 'IncreaseGap' workflow increases the gap by 5 mm. 
Now if Item2 is equal to 0 (meaning the mouse did not cross the gap within 5 min), I want the gap to be reset to its initial distance of 3 cm. 
I already have the workflow that decreases the gap by 5 mm, but I do not know how to decrease the gap by _Counter of gap increases_X 5mm
![image](https://user-images.githubusercontent.com/129788515/229619998-1f0c5bb3-b6da-4544-827f-bba6d94c3474.png)

**Reply by bruno-f-cruz** [reply]
Can you share the actual workflow (`.bonsai`) file? You will need to zip it to attach it as explained in #864.

**Reply by PiezaKake** [reply]
I think I have found a solution, but I have the following issue: I want to compare the output from the 'Accumulate' node, which is a sequence, with an integer. For the comparison, the output from the accumulate node and the integer are fed to a 'Zip' node, and its output is fed to a 'NotEqual' node. The 'Zip' node takes only the first item from the 'Accumulate' node, however I want it to take the last. I have tried to use 
TakeLast(1)', but it did not work. Suggestions?

**Reply by bruno-f-cruz** [reply, marked as answer]
From your description, you might want to use CombineLatest instead of Zip. TakeLast only works if the input sequence has successfully finished.

**Comment by PiezaKake** [comment]
It worked! Thanks!
